File: solutions/csharp/bird-watcher/BirdWatcher.cs
class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek()
    {
        return [0, 2, 5, 3, 7, 8, 4];
    }

    public int Today()
    {
        return birdsPerDay[6];
    }

    public void IncrementTodaysCount()
    {
        birdsPerDay[6]++;
    }

    public bool HasDayWithoutBirds()
    {
        return birdsPerDay.Contains(0);
    }

    public int CountForFirstDays(int numberOfDays)
    {
        int sum = 0;
        if (numberOfDays > 7)
        {
            numberOfDays = 7;
        }
        for (int i = 0; i < numberOfDays; i++)
        {
            sum += birdsPerDay[i];
        }
        return sum;
    }

    public int BusyDays()
    {
        int count = 0;
        for (int i = 0; i < 7; i++)
        {
            if (birdsPerDay[i] >= 5)
            {
                count++;
            }
        }
        return count;
    }
}

File: solutions/julia/bird-watcher/bird-watcher.jl
function today(birds_per_day)
    return birds_per_day[7]
end

function increment_todays_count(birds_per_day)
    birds_per_day[7] += 1
    return birds_per_day
end

function has_day_without_birds(birds_per_day)
    for count in birds_per_day
        if count == 0
            return true
        end
    end
    return false
end

function count_for_first_days(birds_per_day, num_days)
    return sum(birds_per_day[1:num_days])
end

function busy_days(birds_per_day)
    return length(birds_per_day[birds_per_day .>= 5])
end

function average_per_day(week1, week2)
    return (week1 + week2) ./ 2
end
